File: apple/Sources/SignetCore/AttestationResult.swift
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: 2026 Nirapod Labs

import Foundation

/// The result of `getAttestation`. On Apple the Secure Enclave has no per-key
/// hardware attestation. `format` is always `none` and `chain` is always empty.
public struct AttestationResult: Sendable, Equatable {
    /// The attestation wire format.
    public enum Format: String, Sendable, Equatable, CaseIterable {
        /// No attestation is available for this key or platform.
        case none
    }

    public let format: Format
    /// The certificate chain, one DER-encoded certificate per element. Always
    /// empty on Apple, where `format` is always `none`.
    public let chain: [Data]
    public let schemaVersion: Int

    public init(format: Format, chain: [Data] = [], schemaVersion: Int = 1) {
        self.format = format
        self.chain = chain
        self.schemaVersion = schemaVersion
    }
}

File: kmp/signet/src/commonMain/kotlin/org/nirapod/signet/kmp/AttestationResult.kt
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: 2026 Nirapod Labs

package org.nirapod.signet.kmp

/**
 * The result of [Signet.getAttestation]. A hardware key-attestation certificate
 * chain (`androidKeyChain`) is produced only for a key created with an
 * attestation challenge; otherwise `format` is `none` with an empty chain. The
 * challenge is bound at key generation, never here.
 *
 * A plain class, not a data class: `chain` holds raw certificate bytes, where a
 * generated `equals` over array identity would mislead.
 */
public class AttestationResult(
    public val format: Format,
    public val chain: List<ByteArray> = emptyList(),
    public val schemaVersion: Int = 1,
) {
    /** The attestation wire format. */
    public enum class Format {
        /** A hardware key-attestation certificate chain. */
        androidKeyChain,

        /** No attestation is available for this key. */
        none,
    }
}
